README.md

```diff
@@ -4,6 +4,37 @@
 
 This gem is an implementation of the [language server protocol specification](https://microsoft.github.io/language-server-protocol/) for Ruby, used to improve editor features.
 
+# Overview
+
+The intention of Ruby LSP is to provide a fast, accurate, and robust coding environment for Ruby developers, especially for large codebases.
+
+It's part of a [wider Shopify goal](https://github.com/Shopify/vscode-shopify-ruby) to provide a state-of-the-art experience to Ruby developers using modern standards for cross-editor features, documentation and debugging.
+
+It provides many features, including:
+
+* Syntax highlighting
+* Linting and formatting
+* Code folding
+* Selection ranges
+
+When used on typed Sorbet code, additional features are available:
+
+* Quick documentation lookup when hovering over a Ruby or Rails method.
+* Support for Go To Definition.
+
+Planned future features include:
+
+* Context-aware auto-completion and navigation ("Go To Definition") for non-typed code ([prototype](https://github.com/Shopify/ruby-lsp/pull/429))
```

Comment:
What does context-aware mean in this case? Can it be confused with proper typechecking?

Also, let's avoid the usage of "non-typed code". It feels like we're explaining the Ruby LSP from a Sorbet standpoint, when we should just explain the Ruby LSP. It provides features for Ruby code.

Reply:
> What does context-aware mean in this case? Can it be confused with proper typechecking?

I suppose it depends how far https://github.com/Shopify/ruby-lsp/pull/429 goes, but I think at the minimum we'd would be able to auto-complete constant names? Context-aware is maybe a bit vague though, so I'll take that out.